tools/download-test.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+while [[ $# -gt 1 ]]
+do
+  key="$1"
+
+  case "$key" in
+      -t|--tap)
+      TAP="$2"
+      shift # past argument
+      ;;
+      *) # unknown option
+      ;;
+  esac
+  shift # past argument or value
+done
+
+if [ "$DOWNLOAD_LOCATION" == "Nightly" ]; then
+	DOWNLOAD_DIR="nightly"
+else
+	DOWNLOAD_DIR="release"
+fi
+
+LINK=`rsync rsync://unencrypted.nodejs.org/nodejs/$DOWNLOAD_DIR/ | grep $NODE_VERSION | sort -t. -k 1,1n -k 2,2n -k 3,3n | tail -1 | awk '{print $5}'`
+
+if [ -z "$NODE_VERSION" ]; then
+  echo "NODE_VERSION is undefined! Please declare NODE_VERSION"
+  exit 1
+fi
+
+# Remove old files
+[ "$TAP" ] && rm -rf "$TAP"
+rm -rf v*
+
+# Scrape the download server
+rsync -av rsync://unencrypted.nodejs.org/nodejs/"$DOWNLOAD_DIR"/"$LINK" .
+
+num=1
+cat "$LINK"/SHASUMS256.txt | { while read line
+do
+   sha=$(echo "${line}" | awk '{print $1}')
+   file=$(echo "${line}" | awk '{print $2}')
+   echo "Checking shasum for $file"
+   remoteSHA=$(shasum -a 256 "$LINK/$file" | awk '{print $1}')
+   if [ "$remoteSHA" != "$sha" ]; then
+     echo "Error - SHASUMS256 does not match for $file"
+     echo "Expected - $sha"
+     echo "Found - $remoteSHA"
+     [ "${TAP}" ] && echo "not ok $num SHASUMS256 does not match for $file" >> "$TAP"
+     return 1
```

Comment:
Does this work? I'd expect `return: can only 'return' from a function or sourced script`.

Assuming you meant `break`, do we want to bail here? I think we probably want to calculate all the SHAs, otherwise if the first one is wrong it'll bail out, and we won't know how many were wrong.

So maybe just drop this line.

I assume this script will mostly be run through Jenkins, in which case we'll set it to fail on a failed TAP result. If you want to make sure the script returns `1` if a test fails, maybe have a variable:

```bash
  exitStatus=0 # 1 means tests failed.
  while ...; do
    ...
    if [ error ]; then
      exitStatus=1
    fi
  done
}
exit $exitStatus
```
 
If we do this we should make sure the TAP gets parsed in Jenkins even if there are failures though.